src/Evolution/Actions/BackgroundGrVars.hpp

```diff
@@ -0,0 +1,184 @@
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <optional>
+#include <utility>
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "DataStructures/Variables.hpp"
+#include "DataStructures/VariablesTag.hpp"
+#include "Domain/Block.hpp"
+#include "Domain/Creators/Tags/Domain.hpp"
+#include "Domain/Domain.hpp"
+#include "Domain/Structure/BlockId.hpp"
+#include "Domain/Structure/Element.hpp"
+#include "Domain/Structure/ElementId.hpp"
+#include "Domain/Tags.hpp"
+#include "Evolution/Initialization/InitialData.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "Parallel/AlgorithmExecution.hpp"
+#include "Parallel/GlobalCache.hpp"
+#include "ParallelAlgorithms/Initialization/MutateAssign.hpp"
+#include "PointwiseFunctions/AnalyticData/Tags.hpp"
+#include "PointwiseFunctions/InitialDataUtilities/InitialData.hpp"
+#include "PointwiseFunctions/InitialDataUtilities/Tags/InitialData.hpp"
+#include "Time/Tags.hpp"
+#include "Utilities/CallWithDynamicType.hpp"
+#include "Utilities/ErrorHandling/Error.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+namespace evolution::Actions {
+
+/*!
+ * \brief Allocate or assign background general relativity quantities needed
+ * for evolution systems that run on a fixed spacetime e.g. ValenciaDivclean,
+ * ForceFree.
+ *
+ * If a template parameter `used_in_evolution_phase` == true, this action
+ * mutates (updates) the value of background metric variables in box.
+ *
+ * Uses:
+ *  - DataBox:
+ *    * `evolution::initial_data::Tags::InitialData` or
+ *       `::Tags::AnalyticSolutionOrData`
+ *    * `domain::Tags::Coordinates<Dim, Frame::Inertial>`
+ *    * `domain::Tags::Element<Dim>`
+ *    * `domain::Tags::Domain<Dim>`
+ *    * `::Tags::Time`
+ *
+ * DataBox changes:
+ *  - Adds: nothing
+ *  - Removes : nothing
+ *  - Modifies :
+ *    * system::spacetime_variables_tag
+ *    * system::flux_spacetime_variables_tag
+ *    * system::inverse_spatial_metric_tag
+ *
+ * If a template parameter `used_in_evolution_phase` == false, this action can
+ * be used for initialization. Allocate and assign the values of background
+ * metric variables into DataBox.
+ *
+ * Uses:
+ *  - DataBox:
+ *    * `evolution::initial_data::Tags::InitialData` or
+ *       `::Tags::AnalyticSolutionOrData`
+ *    * `domain::Tags::Coordinates<Dim, Frame::Inertial>`
+ *    * `domain::Tags::Mesh<Dim>`
+ *    * `::Tags::Time`
+ *
+ * DataBox changes:
+ *  - Adds:
+ *    * system::spacetime_variables_tag
+ *    * system::flux_spacetime_variables_tag
+ *    * system::inverse_spatial_metric_tag
+ *  - Removes : nothing
+ *  - Modifies : nothing
+ *
+ */
+template <typename System, bool used_in_evolution_phase>
+struct BackgroundGrVars {
+  using simple_tags_from_options = tmpl::list<::Tags::Time>;
+
+  // Collect all the GR quantities used in the templated evolution system
+  using gr_variables_tag = ::Tags::Variables<tmpl::remove_duplicates<
```

Comment:
This line (collection of _all_ the GR related tags) may be unnecessary, would be happy to get comments on this.